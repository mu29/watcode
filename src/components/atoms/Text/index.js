import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { color, fontFamily, fontSize, fontWeight } from 'services/style'

const NormalText = styled.span`
  color: ${color};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${fontWeight};
`

const SmallText = styled.small`
  color: ${color};
  font-family: ${fontFamily};
  font-size: ${fontSize};
  font-weight: ${fontWeight};
`

const Text = ({ small, ...props }) =>
  (small ? <SmallText { ...props } /> : <NormalText { ...props } />)

Text.propTypes = {
  color: PropTypes.string,
  font: PropTypes.string,
  fontSize: PropTypes.number,
  fontWeight: PropTypes.number,
  small: PropTypes.bool,
}

Text.defaultProps = {
  color: 'gray.90',
  font: 'default',
  fontSize: 14,
  fontWeight: 400,
  small: false,
}

export default Text
