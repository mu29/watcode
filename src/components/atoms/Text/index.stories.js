import React from 'react'
import { storiesOf } from '@storybook/react'
import Text from '.'

storiesOf('Atoms/Text', module)
  .add('기본', () => (
    <Text>마음이 여름</Text>
  ))
  .add('크게', () => (
    <Text fontSize={ 32 }>마음이 여름</Text>
  ))
  .add('작게', () => (
    <Text fontSize={ 8 }>마음이 여름</Text>
  ))
  .add('색', () => (
    <Text color="red.default">마음이 여름</Text>
  ))
  .add('폰트', () => (
    <Text font="gothic">마음이 여름</Text>
  ))
