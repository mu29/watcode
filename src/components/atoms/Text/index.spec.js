import React from 'react'
import Text from '.'

const component = (props = {}) => withTheme(<Text { ...props } />)
const wrap = (props = {}) => shallow(component(props)).dive()

describe('<Text />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = render(component())
    expect(wrapper).toMatchSnapshot()
  })

  it('자식 컴포넌트가 들어온 경우 그려준다', () => {
    const wrapper = wrap({ children: 'test' })
    expect(wrapper).toMatchSnapshot()
  })

  it('임의의 속성이 들어온 경우 그려준다', () => {
    const wrapper = wrap({ name: 'foo' })
    expect(wrapper.props().name).toEqual('foo')
  })

  it('span 태그를 기본으로 그려준다', () => {
    const wrapper = wrap()
    expect(wrapper.find('span')).toHaveLength(1)
  })

  it('small 속성이 들어온 경우 smll 태그를 그려준다', () => {
    const wrapper = wrap({ small: true })
    expect(wrapper).toMatchSnapshot()
    expect(wrapper.find('small')).toHaveLength(1)
  })
})
