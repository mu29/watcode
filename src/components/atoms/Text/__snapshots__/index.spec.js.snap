// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Text /> small 속성이 들어온 경우 smll 태그를 그려준다 1`] = `
.c0 {
  color: #424242;
  font-family: 'Spoqa Han Sans','Sans-serif','Helvetica Neue','Nanum Gothic';
  font-size: 0.875rem;
  font-weight: 400;
}

<small
  className="c0"
  color="gray.90"
  fontSize={14}
  fontWeight={400}
/>
`;

exports[`<Text /> 자식 컴포넌트가 들어온 경우 그려준다 1`] = `
.c0 {
  color: #424242;
  font-family: 'Spoqa Han Sans','Sans-serif','Helvetica Neue','Nanum Gothic';
  font-size: 0.875rem;
  font-weight: 400;
}

<span
  className="c0"
  color="gray.90"
  fontSize={14}
  fontWeight={400}
>
  test
</span>
`;

exports[`<Text /> 정상적으로 렌더링된다 1`] = `
.c0 {
  color: #424242;
  font-family: 'Spoqa Han Sans','Sans-serif','Helvetica Neue','Nanum Gothic';
  font-size: 0.875rem;
  font-weight: 400;
}

<span
  class="c0"
  color="gray.90"
  font-size="14"
  font-weight="400"
/>
`;
